models/DataObject/Service.php

```diff
@@ -1633,7 +1633,7 @@ public static function getCalculatedFieldValueForEditMode($object, $params, $dat
                 case DataObject\ClassDefinition\Data\CalculatedValue::CALCULATOR_TYPE_CLASS:
                     $className = $fd->getCalculatorClass();
                     $calculator = Model\DataObject\ClassDefinition\Helper\CalculatorClassResolver::resolveCalculatorClass($className);
-                    if (!$calculator instanceof DataObject\ClassDefinition\CalculatorClassInterface) {
+                    if (!$calculator instanceof ClassDefinition\CalculatorClassInterface) {
```

Comment:
Sorry, I don't see a difference here as both occurrences points to the same interface.

Reply:
For me, when I create a calculated field on data object, The result always returns NULL.
When I traced back the if condition was always getting FALSE, so it return NULL every time. After changing it, the code worked.
May be it could be because of the relative path.

Please try one calculated field and check.